engine/engine.go

```diff
@@ -35,6 +38,7 @@ type Engine struct {
 	prevCacheStats atomic.Value
 
 	dhBatchSize      int
+	dhKeyShard       bool
```

Comment:
Shall we enable this behaviour by default? It's not hurting to have an additional header anyway.

Reply:
I was going to enable it by default in the STI config, but we can do it here also.

Using key sharding is more than just an extra header. It means a separate MergeIndexRequest per multihash.

Reply:
Enabled by default